topic-operator/src/main/java/io/strimzi/operator/topic/BatchingTopicController.java

```diff
@@ -1117,24 +1131,89 @@ private static Collection<AlterConfigOp> buildAlterConfigOps(Reconciliation reco
+     *
+     * <p>This is useful in standalone mode when you have a Kafka service that restricts alter operations
+     * to a subset of all the Kafka topic configurations.</p>
+     *
+     * <p>The default value is "all", which means no restrictions in changing {@code .spec.config}.
+     * The opposite is "none", which can be set to explicitly disable any change.</p>
+     *
+     * @param alterConfigOps Requested alter config operations.
+     */
+    private void skipNonAlterableConfigs(Set<AlterConfigOp> alterConfigOps) {
+        var alterableConfigs = config.alterableTopicConfig();
+        if (alterableConfigs != null && alterConfigOps != null && !alterableConfigs.isEmpty()) {
+            if (alterableConfigs.equalsIgnoreCase("none")) {
+                alterConfigOps.clear();
+            } else if (!alterableConfigs.equalsIgnoreCase("all")) {
+                var alterablePropertySet = Arrays.stream(alterableConfigs.replaceAll("\\s", "").split(","))
+                      .collect(Collectors.toSet());
+                alterConfigOps.removeIf(op -> !alterablePropertySet.contains(op.configEntry().name()));
+            }
+        }
+    }
+
     private static boolean hasConfig(KafkaTopic kt) {
         return kt.getSpec() != null
                 && kt.getSpec().getConfig() != null;
     }
     
     private void updateStatusForSuccess(ReconcilableTopic reconcilableTopic) {
+        List<Condition> conditions = new ArrayList<>();
+        conditions.add(new ConditionBuilder()
+              .withType(TopicOperatorUtil.isPaused(reconcilableTopic.kt()) ? "ReconciliationPaused" : "Ready")
+              .withStatus("True")
+              .withLastTransitionTime(StatusUtils.iso8601Now())
+              .build());
+
+        addNonAlterableConfigsWarning(reconcilableTopic, conditions);
+
         reconcilableTopic.kt().setStatus(
             new KafkaTopicStatusBuilder(reconcilableTopic.kt().getStatus())
-                .withConditions(List.of(new ConditionBuilder()
-                    .withType(TopicOperatorUtil.isPaused(reconcilableTopic.kt()) ? "ReconciliationPaused" : "Ready")
-                    .withStatus("True")
-                    .withLastTransitionTime(StatusUtils.iso8601Now())
-                    .build()))
+                .withConditions(conditions)
             .build());
         updateStatus(reconcilableTopic);
         metrics.successfulReconciliationsCounter(namespace).increment();
     }
-    
+
+    private void addNonAlterableConfigsWarning(ReconcilableTopic reconcilableTopic,
+                                               List<Condition> conditions) {
+        var readOnlyConfigs = new ArrayList<>();
+        var alterableConfigs = config.alterableTopicConfig();
+
+        if (reconcilableTopic != null && reconcilableTopic.kt() != null
+              && hasConfig(reconcilableTopic.kt()) && alterableConfigs != null) {
+            if (alterableConfigs.equalsIgnoreCase("NONE")) {
```

Comment:
again despite the ignore case, why are you using "NONE" here instead of  "none" as before? I would make consistency everywhere, even across the documentation part.